src/main/java/net/snowflake/client/core/HttpUtil.java

```diff
@@ -295,9 +307,18 @@ public static CloseableHttpClient buildHttpClient(
       // only set the proxy settings if they are not null
       // but no value has been specified for nonProxyHosts
       // the route planner will determine whether to use a proxy based on nonProxyHosts value.
+      StringBuilder logBuilder = new StringBuilder();
```

Comment:
Why using string builder? String concatenation has been optimized to string builder under the hood for a long, long time.

Reply:
when it's used in if statement then it's not optimized - it's optimized only for concatenation in one statement

Reply:
Changed it to `+` operator as it will be easier to maintain. This should not be a significant performance hit.